server/src/main/java/com/defold/extender/ExtenderController.java

```diff
@@ -198,17 +206,31 @@ public void buildEngineAsync(HttpServletRequest _request,
             // Regardless of success/fail status, we want to cache the uploaded files
             DataCacheService.DataCacheServiceInfo uploadResultInfo = dataCacheService.cacheFiles(uploadDirectory);
             metricsWriter.measureCacheUpload(uploadResultInfo.cachedFileSize.longValue(), uploadResultInfo.cachedFileCount.intValue());
-            
-            String[] buildEnvDescription = ExtenderUtil.getSdksForPlatform(platform, defoldSdkService.getPlatformSdkMappings(sdkVersion));
-            // Build engine locally or on remote builder
-            if (remoteBuilderEnabled && isRemotePlatform(buildEnvDescription[0], buildEnvDescription[1])) {
-                LOGGER.info("Building engine on remote builder");
-                RemoteInstanceConfig remoteInstanceConfig = getRemoteBuilderConfig(buildEnvDescription[0], buildEnvDescription[1]);
-                this.remoteEngineBuilder.buildAsync(remoteInstanceConfig, uploadDirectory, platform, sdkVersion, jobDirectory, buildDirectory, metricsWriter);
-            } else {
+
+            if (instanceType.equals(InstanceType.BUILDER_ONLY)) {
```

Comment:
If instance marked as "BUILDER_ONLY" - skip request for sdk mappings and start building.